go/vt/vtgate/plan_execute.go

```diff
@@ -140,6 +147,13 @@ func (e *Executor) newExecute(
 			safeSession.RecordWarning(warning)
 		}
 
+		// set the overall query timeout if it is not already set
+		if vcursor.queryTimeout > 0 {
+			timeoutOnce.Do(func() {
```

Comment:
Technically no need for `sync.Once` because there's no concurrency here. I do personally like the use of `sync.Once` even in non-concurrent situations.

Reply:
I can change that. makes sense to just use the variable to validate if it is set or not.